Jellyfin.Networking/Manager/NetworkManager.cs

```diff
@@ -0,0 +1,1283 @@
+#pragma warning restore CS8618 // Non-nullable field is uninitialized.
+
+        /// <summary>
+        /// Event triggered on network changes.
+        /// </summary>
+        public event EventHandler? NetworkChanged;
+
+        /// <summary>
+        /// Gets or sets a value indicating whether testing is taking place.
+        /// </summary>
+        public static string MockNetworkSettings { get; set; } = string.Empty;
+
+        /// <summary>
+        /// Gets or sets a value indicating whether IP6 is enabled.
+        /// </summary>
+        public bool IsIP6Enabled { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether IP4 is enabled.
+        /// </summary>
+        public bool IsIP4Enabled { get; set; }
+
+        /// <inheritdoc/>
+        public NetCollection RemoteAddressFilter { get; private set; }
+
+        /// <summary>
+        /// Gets a value indicating whether is all IPv6 interfaces are trusted as internal.
+        /// </summary>
+        public bool TrustAllIP6Interfaces { get; internal set; }
+
+        /// <summary>
+        /// Gets the Published server override list.
+        /// </summary>
+        public Dictionary<IPNetAddress, string> PublishedServerUrls => _publishedServerUrls;
+
+        /// <inheritdoc/>
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <inheritdoc/>
+        public List<PhysicalAddress> GetMacAddresses()
+        {
+            // Populated in construction - so always has values.
+            lock (_intLock)
+            {
+                return _macAddresses.ToList();
+            }
+        }
+
+        /// <inheritdoc/>
+        public bool IsGatewayInterface(object? addressObj)
+        {
+            var address = addressObj switch
+            {
+                IPAddress addressIp => addressIp,
+                IPObject addressIpObj => addressIpObj.Address,
+                _ => IPAddress.None
+            };
+
+            lock (_intLock)
+            {
+                return _internalInterfaces.Where(i => i.Address.Equals(address) && (i.Tag < 0)).Any();
+            }
+        }
+
+        /// <inheritdoc/>
+        public NetCollection GetLoopbacks()
+        {
+            NetCollection nc = new NetCollection();
+            if (IsIP4Enabled)
+            {
+                nc.Add(IPAddress.Loopback);
+            }
+
+            if (IsIP6Enabled)
+            {
+                nc.Add(IPAddress.IPv6Loopback);
+            }
+
+            return nc;
+        }
+
+        /// <inheritdoc/>
+        public bool IsExcluded(IPAddress ip)
+        {
+            return _excludedSubnets.Contains(ip);
+        }
+
+        /// <inheritdoc/>
+        public bool IsExcluded(EndPoint ip)
+        {
+            if (ip != null)
+            {
+                return _excludedSubnets.Contains(((IPEndPoint)ip).Address);
+            }
+
+            return false;
```

Comment:
I'd rather we re-use function calls between overloads.
```suggestion
            return ip != null && IsExcluded(((IPEndPoint)ip).Address);
```